odoo/addons/base/models/res_users.py

```diff
@@ -774,11 +784,24 @@ def write(self, values):
                 if env.user in self:
                     lazy_property.reset_all(env)
 
-        # clear caches linked to the users
-        if self.ids and 'groups_id' in values:
-            # DLE P139: Calling invalidate_cache on a new, well you lost everything as you wont be able to take it back from the cache
-            # `test_00_equipment_multicompany_user`
-            self.env['ir.model.access'].call_cache_clearing_methods()
+        if 'groups_id' in values:
+            # Do not use `_is_internal` as it relies on the ormcache which is not yet invalidated
+            demoted_users = users_before.filtered(lambda u: self.env.ref("base.group_user") not in u.groups_id)
```

Comment:
Don't forget that `self.env.ref("base.group_user")` always generate a SQL query (exists). Then, it is maybe a good idea to extract it outside of the filtered.

Reply:
Good catch, it's even more queries saved, the test now goes to 16 queries instead of 17 in the added unit test `test_write_groups_id_performance`